app/models/post.rb

```diff
@@ -1,5 +1,15 @@
 class Post < ApplicationRecord
   has_many :comments
   has_many :likes
-  belongs_to :user
+  belongs_to :author, class_name: 'User'
+
+  def count_updater(author_id)
+    user = User.find_by(id: author_id)
+    user.posts_counter = Post.where(author_id:).count
+    user.save
+  end
```

Comment:
Well done on implementing the methods required for this project. 

- [ ] I notice that you have not called this **_count_updater_** method. You can use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#count_updater` method every time a new post has been saved to the database, then make the count_updater` method private. 
- [ ] Also, the code in this method _**count_updater**_ will have performance issues but you can use a more direct and straightforward code as shown below  

See example 👇🏾 
```ruby
# app/models/post.rb
  after_save :update_post_counter

  private

  def update_post_counter
    author.increment!(:posts_counter)
  end
```